### PR TITLE
Fix: Wrong error format when requesting GraphQL with invalid token

### PR DESCRIPTION
**Pull request by hoangmirs**
Resolve https://github.com/nimblehq/nimble-survey-web/issues/51

## What happened
Fix the incorrect error message when an access token is expired
 
## Insight
- Override `doorkeeper_unauthorized_render_options` in `graphql_controller` with the correct message
- Remove redundant code when upgrading GraphQL lib
 

## Proof Of Work
Before:
Altair

After:
Altair

### CODE REVIEW

File: app/controllers/graphql_controller.rb [hunk 1]
```diff
@@ -30,4 +30,31 @@ def ensure_hash(ambiguous_param)
       raise ArgumentError, "Unexpected parameter: #{ambiguous_param}"
     end
   end
+
+  def current_user
+    if doorkeeper_token && !doorkeeper_token.expired?
```
Comment (team-nimblehq) [inline app/controllers/graphql_controller.rb:35]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/d439434a286bd76fead4ad367fe89e1a1cb8da1b/app/controllers/graphql_controller.rb#L35"></span>Favor modifier <code>if</code> usage when having a single-line body. Another good alternative is the usage of control flow <code>&amp;&amp;</code>/<code>||</code>.</td>
    </tr>
  </tbody>
</table>

File: app/controllers/graphql_controller.rb [hunk 2]
```diff
@@ -30,4 +30,31 @@ def ensure_hash(ambiguous_param)
...
+      return User.find(doorkeeper_token.resource_owner_id)
+    end
+
+    raise Doorkeeper::Errors::DoorkeeperError, "Invalid access token"
```
Comment (team-nimblehq) [inline app/controllers/graphql_controller.rb:39]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/d439434a286bd76fead4ad367fe89e1a1cb8da1b/app/controllers/graphql_controller.rb#L39"></span>Prefer single-quoted strings when you don't need string interpolation or special symbols.</td>
    </tr>
  </tbody>
</table>

File: app/controllers/graphql_controller.rb [hunk 3]
```diff
@@ -30,4 +30,31 @@ def ensure_hash(ambiguous_param)
...
+  end
+
+  # :reek:FeatureEnvy
+  def doorkeeper_unauthorized_render_options(error: nil)
```
Comment (team-nimblehq) [inline app/controllers/graphql_controller.rb:43]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/d439434a286bd76fead4ad367fe89e1a1cb8da1b/app/controllers/graphql_controller.rb#L43"></span>Remove unused methods (graphqlcontroller#doorkeeper_unauthorized_render_options)</td>
    </tr>
  </tbody>
</table>

Comment (team-nimblehq) [inline app/controllers/graphql_controller.rb:43]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/d439434a286bd76fead4ad367fe89e1a1cb8da1b/app/controllers/graphql_controller.rb#L43"></span>Method has too many lines. [14/10]</td>
    </tr>
  </tbody>
</table>

File: app/controllers/graphql_controller.rb [hunk 4]
```diff
@@ -30,4 +30,31 @@ def ensure_hash(ambiguous_param)
...
+      json: {
+        errors: [
+          {
+            message: error.description,
+            extensions: {
+              code: error.name,
+              state: error.state
+            }
+          }
+        ],
```
Comment (team-nimblehq) [inline app/controllers/graphql_controller.rb:56]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/d439434a286bd76fead4ad367fe89e1a1cb8da1b/app/controllers/graphql_controller.rb#L56"></span>Avoid comma after the last item of a hash.</td>
    </tr>
  </tbody>
</table>

**Comment by team-nimblehq** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/nimblehq/nimble-survey-web/blob/d439434a286bd76fead4ad367fe89e1a1cb8da1b/app/controllers/graphql_controller.rb#L6">app/controllers/graphql_controller.rb#L6</a> - Remove unused methods (graphqlcontroller#create)</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: app/controllers/graphql_controller.rb [hunk 5]
```diff
@@ -30,4 +30,29 @@ def ensure_hash(ambiguous_param)
...
   end
+
+  def current_user
+    return User.find(doorkeeper_token.resource_owner_id) if doorkeeper_token && !doorkeeper_token.expired?
+
+    raise Doorkeeper::Errors::DoorkeeperError, 'Invalid access token'
+  end
+
+  # :reek:FeatureEnvy
+  def doorkeeper_unauthorized_render_options(error: nil)
```
Comment (team-nimblehq) [inline app/controllers/graphql_controller.rb:41]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/8ba2b428dcbb69ae3682d95ed73a6e3898009224/app/controllers/graphql_controller.rb#L41"></span>Remove unused methods (graphqlcontroller#doorkeeper_unauthorized_render_options)</td>
    </tr>
  </tbody>
</table>

Comment (team-nimblehq) [inline app/controllers/graphql_controller.rb:41]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/8ba2b428dcbb69ae3682d95ed73a6e3898009224/app/controllers/graphql_controller.rb#L41"></span>Method has too many lines. [14/10]</td>
    </tr>
  </tbody>
</table>

File: app/controllers/graphql_controller.rb [hunk 6]
```diff
@@ -30,4 +30,31 @@ def ensure_hash(ambiguous_param)
...
+
+  def current_user
+    return User.find(doorkeeper_token.resource_owner_id) if doorkeeper_token && !doorkeeper_token.expired?
+
+    raise Doorkeeper::Errors::DoorkeeperError, 'Invalid access token'
+  end
+
+  # :reek:FeatureEnvy
+  # rubocop:disable Metrics/MethodLength
+  def doorkeeper_unauthorized_render_options(error: nil)
```
Comment (team-nimblehq) [inline app/controllers/graphql_controller.rb:42]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/7d5f695afb8766158d80c8afb8c895a7e03e8a99/app/controllers/graphql_controller.rb#L42"></span>Remove unused methods (graphqlcontroller#doorkeeper_unauthorized_render_options)</td>
    </tr>
  </tbody>
</table>

File: app/controllers/graphql_controller.rb [hunk 7]
```diff
@@ -30,4 +30,25 @@ def ensure_hash(ambiguous_param)
       raise ArgumentError, "Unexpected parameter: #{ambiguous_param}"
     end
   end
+
+  # :reek:FeatureEnvy
+  # rubocop:disable Metrics/MethodLength
+  def doorkeeper_unauthorized_render_options(error: nil)
```
Comment (team-nimblehq) [inline app/controllers/graphql_controller.rb:36]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/71dc6f6a3fa64cd5be05d7b9e4e8ccc813789ae9/app/controllers/graphql_controller.rb#L36"></span>Remove unused methods (graphqlcontroller#doorkeeper_unauthorized_render_options)</td>
    </tr>
  </tbody>
</table>

**Review by ankitkalia1195 — APPROVED** [review]

**Review by junan — APPROVED** [review]

**Pull request merged**